addons/api/addon/utils/flatten-nested-object.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+/**
+ * Flattens a nested object into a single-level object.
+ * @param obj
+ * @param result
+ * @param parentPath
+ * @returns {Object}
+ */
+
+export const flattenObject = (obj, result = {}, parentPath = '') => {
```

Comment:
None blocking comment: I was trying to understand what `parentPath` is, now I do, asking your opinion, do you think `parentKey` can be more descriptive?

My thinking:
if `b: { c: 2, d: 3 },` transforms to `{'b.c': 2, 'b.d': 3}` seems more `parentKey` is b while `key` is c or d, dependending the case

Reply:
yeah I can change to key, I think that was the name I originally had lol